include/seastar/core/abort_source.hh

```diff
@@ -48,7 +49,7 @@ public:
 /// Facility to communicate a cancellation request to a fiber.
 /// Callbacks can be registered with the \c abort_source, which are called
 /// atomically with a call to request_abort().
-class abort_source {
+class abort_source : public enable_shared_from_this<abort_source> {
```

Comment:
Why not make the standard abort_source "shareable" by making the second abort() no-op? It already allows multiple subscriptions, so is shared by consumers.

Reply:
In any case, please add a respective unit test.